Modules/Core/Common/test/itkFixedArrayTest.cxx

```diff
@@ -31,7 +31,7 @@ Set_c_Array(int x[3])
 }
 
 void
-Print_Array(itk::FixedArray<int, 3> x, std::ostream & os)
+Print_Array(const itk::FixedArray<int, 3> & x, std::ostream & os)
```

Comment:
In this particular case, either way is fine by me. `itk::FixedArray<int, 3>` can be copied very quickly, as it is trivially copy-able. So the original pass-by-value does not involve dynamic memory allocation or the execution of a hand-written copy-constructor.

In general, pass-by-value (rather than pass-by-const-reference) has the advantage that it is more thread-safe. With pass-by-const-reference, the caller is responsible for keeping the object alive and unmodified, while the function is running. So pass-by-value is often simpler.

However, when copying is more expensive (as with an arbitrary `std::vector`), it's often beneficial to go for pass-by-const-reference. Then again, sometimes move-semantics is even better...

So as always, each approach has its pro's and con's  🤔

Reply:
Once we would have cppcheck on the CI, would it still be possible to _allow_ pass-by-value (rather than pass-by-const-reference) in specific cases? For object types that can be copied very quickly. Or for specific functions?

Because as I wrote before, there _are_ cases where pass-by-value is preferable.

Reply:
I don't think it warns about every single pass-by-value under the sun, only for some classes where it usually makes sense, like std::string.

Reply:
Indeed I see now that cppcheck does not warn for every pass-by-value (under the 🌞). Looks like it does not warn when you pass an empty struct by value.  However, it does warn for `itk::FixedArray`. Looks like it even warns for very small ones, for example  `itk::FixedArray<int, 1>` and `itk::FixedArray<char, 1>`. While passing those two by-value is probably _faster_ than passing them by reference!

I'm not opposed to this specific code change, it's just a test anyway. But I just doubt if we should _always_ enforce pass-by-const-reference whenever cppcheck tells us to do so.

Reply:
FYI, I just posted at https://sourceforge.net/p/cppcheck/discussion/general/thread/cce6de4bfd/ "How to deal with passByValue (CWE: 398) warnings?"